Work in progress: Control objectives

leovdoo opened the pull request:
Edits for grammar and readability, added motivation in most sections.

scriptingxss commented on en/V4-Communication_Requirements.md:
```diff
@@ -2,15 +2,13 @@
 ## Control Objective
 
-Ensure that a verified application satisfies the following high level requirements:
+Devices use network communication to exchange data and receive commands with its ecosystem. So that the different parties can trust the contents of communications, they need to be protected, ensuring authenticity of the parties, integrity against malicious changes, and confidentiality against information leakage. In practice, this translates to deploying up-to-date communication protocols and con ...
```
Proposed edits below:

Devices use network communication to exchange data and receive commands within their ecosystem. So that the different parties can trust the contents of communications, they need to be protected, ensuring the authenticity of parties, integrity against malicious changes, and confidentiality against information leakage. In practice, this translates to deploying up-to-date communication protocols and configuring their security features, including cryptography. Since industry guidelines on secure TLS, Bluetooth, and Wi-Fi change frequently, configurations should be periodically reviewed to ensure that communications security is always effective.

scriptingxss commented on en/V5-Hardware_Platform_Requirements.md:
```diff
@@ -2,9 +2,9 @@
+Hardware is more difficult and more costly to compromise and subvert than software. Therefore hardware security can provide a robust foundation for embedded device security. On the other hand, hardware that contains backdoors or undocumented debug features can completely compromise the security of the entire device. 
 
-The purpose of the controls listed in this section is to ensure that as long as hardware is available for secure configuration, it is been configured in the maximum possible secure way. This includes disabling or securing debug interfaces, setting up all existing alarms and sensors mechanism to combat tampering, use anti-cloning hardware protection, like OTP fuses, use MMU (Memory Management Unit ...
+The purpose of the controls listed in this section is to ensure that as long as hardware is available for secure configuration, it is been configured in the most secure way possible. This includes disabling or securing debug interfaces, setting up all existing alarms and sensors mechanisms to combat tampering, using anti-cloning hardware protection such as OTP fuses, and the use of the MMU (Memor ...
```
Minor edit as noted in a previous comment - Please use "in this chapter" versus "in this section"

Change "sensors" to singular "sensor mechanisms"

scriptingxss reviewed (changes requested):
Added comments and suggestions. Thanks!

OWASP-ISVS-Gibook commented on en/V5-Hardware_Platform_Requirements.md:
```diff
@@ -2,9 +2,9 @@
 
 ## Control Objective
 
-The secure hardware configuration is one of the fundamentals of the embedded device security. The device code can be written and configured in the most secure way, but it cannot help as long as hardware has backdoors in a form of open debug ports or other undocumented debug features.
```
Small comment: this chapter provides the requirements for hardware platform such that a secure hardware configuration can be guaranteed. For example, 
- 3.1.4 discusses correctly configuring secure boot. 5.1.2 requires that the platform supports this. 
- 5.1.1 requires that the platform supports disabling debug interface. 1.2.4 requires that it is done so in production.
Can you take up this nuance in the description? 

Thanks!

cbassem commented on en/V5-Hardware_Platform_Requirements.md:
```diff
@@ -2,9 +2,9 @@
 
 ## Control Objective
```
Small comment: this chapter provides the requirements for hardware platform such that a secure hardware configuration can be guaranteed. For example,

3.1.4 discusses correctly configuring secure boot. 5.1.2 requires that the platform supports this.
5.1.1 requires that the platform supports disabling debug interface. 1.2.4 requires that it is done so in production.
Can you take up this nuance in the description?
Thanks!

leovdoo replied:
I'm adding this, but we don't have the same kind of summary in other sections.

leovdoo commented on en/V2-User_Space_Application_Requirements.md:
```diff
@@ -2,11 +2,15 @@
-The purpose of the controls listed in this section is to ensure the authenticity,  authorization of IoT system users and machines, sensitive data protection by using the best security practices, like strong unique secure identity, user role segregation, sensitive data and credentials protection and fair private information treatment, concept of least privilege, and best cryptography algorithms.  
+The purpose of the controls listed in this section is to ensure secure access to an IoT system by users and machines and to protect sensitive data by using security best practices. 
+
+Authentication and authorization are necessary to secure access. Relevant controls strong unique secure identity, user role segregation, and the concept of least privilege. Authentication is the act of establishing, or confirming, the identity of someone (or something) as authentic, as a basis to believe that claims made by a person or about a device are correct and resistant to impersonation. Ad ...
```
Yes, fixing

leovdoo commented on en/V2-User_Space_Application_Requirements.md:
```diff
@@ -2,11 +2,15 @@
+
+Authentication and authorization are necessary to secure access. Relevant controls strong unique secure identity, user role segregation, and the concept of least privilege. Authentication is the act of establishing, or confirming, the identity of someone (or something) as authentic, as a basis to believe that claims made by a person or about a device are correct and resistant to impersonation. Ad ...
+
+Protection for sensitive data, including credentials, and fair treatment of private information, are necessary to ensure secure use of system resources, including any files containing data or code, and the contents of memories.
```
Fixing: Protection of sensitive data including credentials, and fair treatment of private information, are necessary to ensure secure use of system resources, such as files containing data or code, and the contents of memory.

leovdoo commented on en/V2-User_Space_Application_Requirements.md:
```diff
@@ -2,11 +2,15 @@
 ## Control Objective
 
-The purpose of the controls listed in this section is to ensure the authenticity,  authorization of IoT system users and machines, sensitive data protection by using the best security practices, like strong unique secure identity, user role segregation, sensitive data and credentials protection and fair private information treatment, concept of least privilege, and best cryptography algorithms.  
+The purpose of the controls listed in this section is to ensure secure access to an IoT system by users and machines and to protect sensitive data by using security best practices. 
```
Fixing

leovdoo commented:
@scriptingxss , I've addressed your comments, and solved the newline issue as far as I can tell. Please review, if it looks alright, then I think we can merge.

leovdoo commented:
If any conflicts remain, you can safely resolve them with the version in master.
